barretenberg/cpp/pil/avm/avm_alu.pil

```diff
@@ -264,4 +264,195 @@ namespace avm_alu(256);
     #[ALU_OP_EQ]
     op_eq * (DIFF * (ic * (1 - op_eq_diff_inv) + op_eq_diff_inv) - 1 + ic) = 0;
 
-
+    // ========= LT/LTE Operation Constraints ===============================
+
+    pol commit op_lt;
+    pol commit op_lte;
+
+    // Used for perm check to main, designates if a comparison operation is being used.
+    pol commit cmp_sel; 
+    cmp_sel = op_lt + op_lte;
+
+    // ic is restricted to be 1 or 0 if we are performing a comparison
+    cmp_sel * (ic * (1 - ic)) = 0;
+
+   
+
+    // There are two routines that we utilise as part of this LT/LTE check
+    // (1) Decomposition into two 128-bit limbs, lo and hi respectively and a borrow (1 or 0);
+    // (2) 128 bit-range checks when checking an arithmetic operation has not overflowed the field.
+
+    // ========= COMPARISON OPERATION - EXPLANATIONS =================================================
+    // To simplify the comparison circuit, we implement a GreaterThan(GT) circuit. This is ideal since
+    // if we need a LT operation, we just swap the inputs and if we need the LTE operation, we just NOT the GT constraint
+    // Given the inputs x, y and q where x & y are integers in the range [0,...,p-1] and q is the boolean result to the query (x > y).
+    // Then there are two scenarios:
+    //    (1) (x > y) -> x - y - 1 = result, where 0 <= result. i.e. the result does not underflow the field.
+    //    (2)!(x > y) -> (x <= y) = y - x = result, where the same applies as above.
+
+    // These conditions can be combined with the GT constraint, q (that x > y) as follows:
+    // (x - y - 1) * q + (y - x) (1 - q) = result
+    pol commit input_ia;
```

Comment:
input_ia and input_ib do not have to be commited. I think they could be migrated to tmp value.
pol INPUT_IA = ...
pol INPUT_B = ...